histoqc/BaseImage.py

```diff
@@ -98,11 +108,30 @@ def getImgThumb(self, dim):
         key = "img_" + str(dim)
         if key not in self:
             osh = self["os_handle"]
+            xc=yc=0
+            if self["bounding_box"] == True:
+                try:
+                    xc=int(osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_X,'NA'))
+                    yc=int(osh.properties.get(openslide.PROPERTY_NAME_BOUNDS_Y,'NA'))
+                except:
+                    logging.error(f"{self['filename']}: could not find bounding box coordinates")
+                    return -1
             if dim.replace(".", "0", 1).isdigit(): #check to see if dim is a number
                 dim = float(dim)
                 if dim < 1 and not dim.is_integer():  # specifying a downscale factor from base
                     new_dim = np.asarray(osh.dimensions) * dim
-                    self[key] = np.array(osh.get_thumbnail(new_dim))
+                    if self["bounding_box"] == True:
+                        downsample = max(*(dim / thumb for dim, thumb in zip(osh.dimensions, new_dim)))
+                        level = osh.get_best_level_for_downsample(downsample)
```

Comment:
shouldn't this get_best_level_for_downsample be addressed the same way as the corrected version you produced below? likely want to turn that into a function that wraps around the get_best_level_for_downsample  function in all places?